src/lib.rs

```diff
@@ -189,9 +188,34 @@ unsafe extern "C" fn reset_handler() -> ! {
     r0::zero_bss(&mut _sbss, &mut _ebss);
     r0::init_data(&mut _sdata, &mut _edata, &_sidata);
 
-    // Neither `argc` or `argv` make sense in bare metal context so we just
-    // stub them
-    main(0, ::core::ptr::null());
+    match () {
+        #[cfg(not(has_fpu))]
+        () => {
+            // Neither `argc` or `argv` make sense in bare metal context so we just
+            // stub them
+            main(0, ::core::ptr::null());
+        }
+        #[cfg(has_fpu)]
+        () => {
+            // NOTE(safe) no exception / interrupt that also accesses the FPU
+            // can occur here
```

Comment:
I probably didn't word this correctly. I'm not referring to the "enable the FPU" operation as being unsafe, but rather that a RMW operation happens here (on FPU related registers but that doesn't matter) and that it's not 100% certain that the RMW operation will be executed atomically because it's *not* executed within a critical section so a data race, by definition, is possible here.

My comment tries to argue that no user defined interrupt / exception handler, which may also access the same memory location (SCB.CPACR), is likely to preempt this operation.